apps/api/src/app/events/usecases/create-notification/create-notification.command.ts

```diff
@@ -0,0 +1,34 @@
+// TODO: We shouldn't be importing from DAL here. Needs big refactor throughout monorepo.
+import { NotificationTemplateEntity, SubscriberEntity } from '@novu/dal';
+import { ISubscribersDefine } from '@novu/shared';
+import { IsDefined, IsString, IsOptional } from 'class-validator';
+
+import { EnvironmentWithUserCommand } from '../../../shared/commands/project.command';
+
+export class CreateNotificationCommand extends EnvironmentWithUserCommand {
+  @IsDefined()
+  @IsString()
+  identifier: string;
+
+  @IsDefined()
+  overrides: Record<string, Record<string, unknown>>;
+
+  @IsDefined()
+  payload: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
I think here we would need to also make shareable the interface `ITriggerPayload` so we can remove all any set for payload property. But not 100% sure if it will be right interface.

Reply:
Soon to be fixed.